src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs

```diff
@@ -86,9 +86,9 @@ public OpenApiDocument GetSwagger(string documentName, string host = null, strin
 
             var applicableApiDescriptions = _apiDescriptionsProvider.ApiDescriptionGroups.Items
                 .SelectMany(group => group.Items)
-                .Where(apiDesc => !(_options.IgnoreObsoleteActions && apiDesc.CustomAttributes().OfType<ObsoleteAttribute>().Any()))
-                .Where(apiDesc => !apiDesc.CustomAttributes().OfType<SwaggerIgnoreAttribute>().Any())
-                .Where(apiDesc => _options.DocInclusionPredicate(documentName, apiDesc));
+                .Where(apiDesc => !(_options.IgnoreObsoleteActions && apiDesc.CustomAttributes().OfType<ObsoleteAttribute>().Any()) &&
```

Comment:
If this were turned into a "full" lambda method block I think it would improve the readability over this long expression, and then you could declare a local to store the attributes into for the two different queries on them.

Reply:
done